tests/semantic_analyser.cpp

```diff
@@ -4304,6 +4306,86 @@ stdin:1:34-47: ERROR: Type mismatch for $a: trying to assign value of type 'stru
 BEGIN { let $a: struct bad_task; $a = *curtask; }
                                  ~~~~~~~~~~~~~
 )");
+
+  test_error(R"(BEGIN { $x = 2; if (1) { let $x; } })", R"(
+stdin:1:26-32: ERROR: Variable declarations need to occur before variable usage or assignment. Variable: $x
+BEGIN { $x = 2; if (1) { let $x; } }
+                         ~~~~~~
+)");
+}
+
+TEST(semantic_analyser, block_scoping)
+{
+  // if/else
+  test("BEGIN { $a = 1; if (1) { $b = 2; print(($a, $b)); } }");
+  test("BEGIN { $a = 1; if (1) { print(($a)); $b = 2; if (1) { print(($a, "
+       "$b)); } else { print(($a, $b)); } } }");
```

Comment:
I know it's just a test but could we have a nicer indentation of programs which span multiple lines? Same for the other test cases in this block.